components/raftstore/src/store/peer.rs

```diff
@@ -2489,6 +2538,11 @@ where
 
             let persist_index = self.raft_group.raft.raft_log.persisted;
             self.mut_store().update_cache_persisted(persist_index);
+
+            if let Some(ForceLeaderState::ForceLeader { .. }) = self.force_leader {
+                // forward commit index, the committed entries will be applied in the next raft base tick round
+                self.force_forward_commit_index();
```

Comment:
`force_forward_commit_index` only forward commit index if all alive peers have also persisted the logs, if the leader failed to forward, then it may not get a second chance to retry `force_forward_commit_index` here

Reply:
you're right, should check it in base tick too.